src/sources/kafka.rs

```diff
@@ -179,6 +183,23 @@ fn kafka_source(
                                 log.insert(offset_key, Value::from(msg.offset()));
                             }
 
+                            if let Some(headers_key) = &headers_key {
+                                let mut headers_map = BTreeMap::new();
+                                if let Some(headers) = msg.headers() {
+                                    // Using index-based for loop because rdkafka's `Headers` trait
+                                    // does not provide Iterator-based API
```

Comment:
Ugh, that's… disappointing, but yes that does require this kind of interface. :disappointed: